docs/docs/spec.md

```diff
@@ -270,49 +273,50 @@ following cases: \[[#17]\]
+    any of the following:
 
-> These locations all fit under the umbrella of "implementation code."
-> Implementation code may use types that contain type arguments, wildcard
-> bounds, and array component types, which would be recognized locations if not
-> for the exceptions defined by this section.
+    -   a local variable
 
--   a local variable type
--   the type in a cast or `instanceof` expression
--   an array or object creation expression (including via a member reference)
--   an explicit type argument supplied to a generic method or constructor
-    (including via a member reference) or to an instance creation expression for
-    a generic class
+    -   the type in a cast of `instanceof` expression
 
-> In practice, we anticipate that tools will treat types (and their annotations)
-> in *most* of the above locations much like they treat types in other
-> locations. Still, this spec does not concern itself with implementation code:
-> We believe that the most important domain for us to focus on is that of APIs.
+    -   an object creation expression
+
+    > In practice, we anticipate that tools will be able to infer the nullness of
+    > these parts of implementation code without explicit annotation, and so we do
+    > not recognize nullness annotations at those locations in this specification.
+    >
+    > For example, the annotation is unrecognized in the following cases:
+    > -   `@Nullable List<String> strings = ...`
+    > -   `String @Nullable [] strings = ...`
+    > -   `(@Nullable List<String>) foo`
+    > -   `new @Nullable ArrayList<String>()`
+    >
+    > But the following are all recognized:
+    > -   `List<@Nullable String> strings = ...`
+    > -   `@Nullable String[] strings = ...`
+    > -   `(List<@Nullable String>) foo`
+    > -   `new ArrayList<@Nullable String>()`
+
+-   type arguments of a receiver parameter's type
 
 All locations that are not explicitly listed as recognized are unrecognized.
 
-> Other notable unrecognized annotations include: \[[#17]\]
->
-> Some additional intrinsically non-nullable locations:
->
-> -   supertype in a class declaration
-> -   thrown exception type
-> -   exception parameter type
-> -   enum constant declaration
-> -   receiver parameter type
->
-> Some other locations that individual tools are more likely to assign semantics
-> to:
->
-> -   a class declaration \[[#7]\]: For example, the annotation in `public
->     @Nullable class Foo {}` is in an unrecognized location.
-> -   a type-parameter declaration or a wildcard *itself* \[[#19], [#31]\]
-> -   any [type component] of a receiver parameter type \[[#157]\]
+> Other notable unrecognized annotations include:
+> -   class declaration: For example, the annotation in `public @Nullable
+>     class Foo {}` is in an unrecognized location.
+> -   type-parameter declaration or a wildcard *itself*
+> -   any [type component] of a receiver parameter type
```

Comment:
This is now covered by "type arguments of a receiver parameter's type" above, so we can remove it.